schematic/models/validate_attribute.py

```diff
@@ -1,1053 +1,1101 @@
+from schematic.utils.validate_utils import (
+    comma_separated_list_regex,
+    parse_str_series_to_list,
+    np_array_to_str_list,
+    iterable_to_str_list,
+    rule_in_rule_list,
+)
+
+from synapseclient.core.exceptions import SynapseNoCredentialsError
+
+logger = logging.getLogger(__name__)
+
+
+class GenerateError:
+    def generate_schema_error(
+        row_num: str,
+        attribute_name: str,
+        error_message: str,
+        invalid_entry: str,
+        dmge: DataModelGraphExplorer,
+    ) -> Tuple[list[list[str]], list[list[str]]]:
+        """
+        Purpose: Process error messages generated from schema
+        Input:
+            - row_num: the row the error occurred on.
+            - attribute_name: the attribute the error occurred on.
+            - error_message: Error message
+            - invalid_entry: The value that caused the error
+            - dmge: DataModelGraphExplorer object
+        """
+
+        error_list, warning_list = GenerateError.raise_and_store_message(
+            dmge=dmge,
+            val_rule="schema",
+            error_row=row_num,
+            error_col=attribute_name,
+            error_message=error_message,
+            error_val=invalid_entry,
+        )
+
+        return error_list, warning_list
+
+    def generate_list_error(
+        list_string: str,
+        row_num: str,
+        attribute_name: str,
+        list_error: str,
+        invalid_entry: str,
+        dmge: DataModelGraphExplorer,
+        val_rule: str,
+    ) -> Tuple[list[list[str]], list[list[str]]]:
+        """
+        Purpose:
+            If an error is found in the string formatting, detect and record
+            an error message.
+        Input:
+            - list_string: the user input list, that is represented as a string.
+            - row_num: the row the error occurred on.
+            - attribute_name: the attribute the error occurred on.
+            - list_error: the type of list error that occurred.
+            - invalid_entry: The value that caused the error
+            - dmge: DataModelGraphExplorer object
+            - val_rule: validation rule str, defined in the schema.
+        Returns
+        Errors: list[str] Error details for further storage.
+        warnings: list[str] Warning details for further storage.
+        """
+
+        if list_error == "not_comma_delimited":
+            error_message = (
+                f"For attribute {attribute_name} in row {row_num} it does not "
+                f"appear as if you provided a comma delimited string. Please check "
+                f"your entry ('{list_string}'') and try again."
+            )
+        elif list_error == "not_a_string":
+            error_message = (
+                f"For attribute {attribute_name} in row {row_num} it does not "
+                f"appear as if you provided a string. Please check "
+                f"your entry ('{list_string}'') and try again."
+            )

```

Comment:
I believe the cases are covered so that nothing else could be passed in, but I'll add a check to cover the case were `None` is received.

Reply:
So what i would do is at the top of this file: 
`from typing import Literal`

Then on line 64 replace:
`list_error: str,`
with
`list_error: Literal["not_comma_delimited", "not_a_string"]`

Then in the future once we've are checking everything with mypy this will catch if anythign calls this with not one of those values.

Reply:
Thank you! I'll make that change